File: 1-10/P2.java
import java.util.Scanner;

public class P2 {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter String : ");
        String str=sc.nextLine();
        checkPalinString(str);
    }

    public static void checkPalinString(String str){
        int i, j;
        for (i = str.length() - 1, j = 0; i >= 0 && j < str.length(); i--, j++) {
            if (str.charAt(i) != str.charAt(j)) {
                System.out.println("String is not Palindrome");
                return;
            }
        }
        System.out.println("String is Palindrom");
    }
}

File: 1-10/P2.c
#include <stdio.h>

void main()
{

	char str[100], rev[100];
	int len, i, j;
	boolean flag = false;

	printf("Enter a string: ");
	gets(str);

	len = strlen(str);

	for (i = 0; i < len / 2; i++)
	{
		if (str.charAt(i) != str.charAt(len - i - 1))
		{
			flag = true;
			break;
		}
	}

	if (flag)
	{
		printf("string is Palindrome");
	}
	else
		printf("string is Palindrome");
}
